test/util/common.js

```diff
@@ -53,6 +53,35 @@ module.exports.eraseJlinkDevice = device => {
     });
 };
 
+/**
+ * Completely erase the device with `recover` functionality, then program it
+ * with bootloader hex file.
+ *
+ * @param {Object} device Device object from nrf-device-lister.
+ * @param {string} filename Filename of the bootloader hex file.
+ * @returns {Promise<Object>} Resolves with the device object if successful.
+ */
+module.exports.programBootloaderJlinkDevice = (device, filename) => {
+    return new Promise((resolve, reject) => {
+        const serialNumber = parseInt(device.serialNumber, 10);
+        nrfjprog.recover(serialNumber, error => {
+            if (error) {
+                reject(error);
+            } else {
+                const options = {
+                    chip_erase_mode: nrfjprog.ERASE_ALL,
+                    qspi_erase_mode: nrfjprog.ERASE_ALL,
```

Comment:
Do we need to set qspi_erase_mode?

Reply:
I don't exactly know, but this made the bootloader upgradeable. It's only a test case, not exposed in the final module.